### DISCUSSION TITLE
5G Router/Hotspot Ping Time?

### DISCUSSION THREAD

**Discussion by A5005** [jamulussoftware/jamulus#1642] [General]
5G routers are already available on the market. They must have an incredibly short ping time. Has anyone tried that with Jamulus? Or the slightly less efficient variant: 5G mobile Phone Hotspot WLAN to Computer?  In Germany there is currently a 30-day test offer from Vodafone: The 5G GigaCube (also has LAN connection). https://zuhauseplus.vodafone.de/gigacube/ 
Maybe I'll try it out and report here  if no one answers that 5G is not recommended.

**Comment by dcorson-ticino-com** [comment]
This is certainly very interesting, but at least 2 factors make me skeptical until proven otherwise.
- To get the best, as advertised, connectivity, one must be very close to a 5G antenna.
- Most of the 5G installations, at least up 'til now, are a software update on 4G hardware. 
Please keep us up-to-date on any tests you may make, if it really does work well 5G could enable many people.

EDIT: Don't forget to use a cable connection between the router and your end device.  Using a 4G router and a cabled connection is just usable, add wifi to the equation and all hope is gone.

**Reply by gene96817** [reply]
I am also skeptical. The 5G router will help only if it is communicating with a cell using the high-band spectrum (i.e. a small cell).  Your 5G router is only the first hop to the Jamulus server. I am most interested in how much overall delay can be reduced. by that first hop.

> Most of the 5G installations, at least up 'til now, are a software update on 4G hardware.

This would then be a 5G cell using the mid-band frequencies. The much-promoted 5G speeds assume using the high-band spectrum.

> To get the best, as advertised, connectivity, one must be very close to a 5G antenna.

Is this a symptom of being connected to a 5G small cell (again, meaning a cell using the high-band spectrum)?

**Comment by ignotus666** [comment]
My router and laptop both support 5G wifi. I tested it with Jamulus and the result was surprisingly good. Far better than "regular" wifi and only a marginally worse ping than over an ethernet cable. For some stupid reason my router won't do ethernet over USB and my laptop doesn't have an ethernet port, so if I didn't have a PC with one it would've been the only option. I was close to the router though.

**Reply by npostavs** [reply]
>  For some stupid reason my router won't do ethernet over USB and my laptop doesn't have an ethernet port

I think the usual solution is a USB-to-ethernet adapter which goes into the laptop. I've never seen anyone suggest ethernet over USB before.

**Reply by ignotus666** [reply]
My previous router could do ethernet over USB. This one has the USB ports but the provider disabled them via firmware.

**Reply by gene96817** [reply]
Increasingly notebooks provide only USB(-C)  ports. A USB-to-ethernet adapter is the common solution. 

I am curious what the numbers are behind "marginally worst ping". Does that mean it adds 4 ms of ping?

**Reply by ignotus666** [reply]
I did the test some time ago and can't remember the precise number. I'll do it again tonight and report back.

**Reply by ignotus666** [reply]
Just tested: ping time is 1-2 ms higher than with a wired ethernet connection. Tested by hooking up to the same server (not mine) with my PC (wired ethernet) PC and laptop (5G WiFi). The difference is that I have to raise the jitter buffers by a notch or two for comparably clean audio. So with 5G WiFi I get an overall latency that is about 5-6 ms higher than wired.

**Reply by npostavs** [reply]
What do you mean by 5G Wifi exactly? Is that 5GHz? https://www.howtogeek.com/398620/whats-the-difference-between-5g-and-5ghz-wi-fi/

**Reply by dcorson-ticino-com** [reply]
That is my question too:
Are we talking about 5Ghz wifi or 5th generation GSM ?

**Reply by ignotus666** [reply]
Ahhh, there you go, hehe - my bad. I was talking about 5 GHz WiFi and I now realise the OP was probably talking about 5G the successor of 4G... Sorry for the misleading comments!

**Reply by gene96817** [reply]
Now that it is clear we are talking about 5GHz Wi-Fi, the latency will always be larger than wired (Ethernet) because of the wireless media access protocol. Further, the latest version of Wi-Fi (marketed as WiFi6) won't reduce the ping time because it has to be backward compatible with earlier versions.

**Comment by ann0see** [comment]
My choir bought a 5G router which unfortunately doesn’t seem to get 5G signal inside our church. I‘m quite sure we can test it again soon.

I did succeed to sing over 4G with another router by Deutsche Telekom and the quality is useable.

**Reply by JamulusUser** [reply]
maybe it is a good idea to place the router on the church tower to receive 5G ... just kidding ;-)

**Reply by ann0see** [reply]
… which would mean we need a very long Ethernet cable ;-). 

The church is quite near to a cell tower. Maybe we could place the router near an open window/door.

**Reply by dcorson-ticino-com** [reply]
IMHO 5G is going to end up being an expensive disappointment for both providers and users.
Under ideal circumstances really neat, in the real world, ho hum.

**Reply by gene96817** [reply]
I assume we are talking about small cell 5G in the high-frequency band. All. the 5G buzz is with the high-frequency band. In this case, you want to maximize the line-of-sight between. your router in the open window with the cell tower. (The mid-band 5G frequencies have properties similar to 4G.)

**Comment by gene96817** [comment]
5G has a lot more trouble penetrating buildings (and foliage).
The top of the church tower is really a good place for 5G.    :P

**Comment by ann0see** [comment]
OK. Just tried the 5G router near a window and we could't get a stable signal. Overall delay is quite high around 80ms

**Comment by ghost** [comment]
The 5G GigaCube 'maximum of 500 mega-bits/second = 50x10^6 octets per second' ; uses '5G cell service' and 'Wi-Fi 6' or (Wi-Fi 1 to 5 for older devices).
- So use a fiber-optic cable service, and CAT6 ethernet modem for more Jamin' goodness.
- But Wi-Fi 6 is the best wi-fi so far. Wi-Fi uses either/both 2.4 GHz and 5 GHz radiowave carrier frequencies.
- 5G is the best cell phone service so far.

**Comment by ann0see** [comment]
Ok. Finally had a 5G signal! Ping is unstable and can be as low as 15ms or as high as 40ms

**Comment by gene96817** [comment]
What you observe with 5G applies to all mobile data service. You are sharing the capacity of the cell site. Your ping times and effective throughput will depend on the number of subscribers using the cell site. A private Wi-Fi access point has similar characteristics. Even if you control who uses your access point. You cannot control who is sharing the wi-fi channel (through another access point).

**Reply by ann0see** [reply]
Yes. Sure. Unfortunately sometimes that's the only way to get a connection to the web.

I'm looking forward to getting a long ethernet cable for our church ;-). It should be possible to get a reliable connection from another building. In the mean time 5G is our only solution.

**Reply by gene96817** [reply]
As you work on a long ethernet cable, sometimes the distance exceeds the "official" cable length. Don't forget you can use an inexpensive hub to connect two long cables. The extra delay should be under 1 ms.
